File: Supercomparator Springboot API/src/main/java/com/tfg/supercomparator/domain/ProductHistory.java
package com.tfg.supercomparator.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;

import java.time.LocalDate;
import java.util.Map;

@Data
@AllArgsConstructor
@NoArgsConstructor
public abstract class ProductHistory {
    @Id
    private String name;
    private Map<LocalDate, Float> history;
}

File: SuperComparator/app/src/main/java/com/tfg/supercomparator/domain/modules/model/product/ProductHistory.kt
package com.tfg.supercomparator.domain.modules.model.product

import java.time.LocalDate

data class ProductHistory(
    private val history: Map<String, Float>,
) {
    companion object {
        fun ProductHistory.mapToLocalDateString(): Map<LocalDate, Float> {
            return this.history.mapKeys {
                LocalDate.parse(it.key)
            }
        }
    }
}

